config/config.go

```diff
@@ -664,6 +671,43 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return checkOverflow(c.XXX, "kubernetes_sd_config")
 }
 
+// EC2SDConfig is the configuration for EC2 based service discovery.
+type EC2SDConfig struct {
+	Region          string                    `yaml:"region"`
+	Access          string                    `yaml:"access,omitempty"`
+	Secret          string                    `yaml:"secret,omitempty"`
+	RefreshInterval Duration                  `yaml:"refresh_interval,omitempty"`
+	Port            int                       `yaml:"port"`
+	Tags            map[string]string         `yaml:"tags,omitempty"`
```

Comment:
I still don't understand how this plays into the public/private IP question, since which IP to use as the target doesn't come into play after relabeling. Or does it and I just haven't seen how that works?